components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -221,6 +224,29 @@ func InjectOAuthProxy(notebook *nbv1.Notebook, oauth OAuthConfig) error {
 	return nil
 }
 
+func (w *NotebookWebhook) ClusterWideProxyIsEnabled() bool {
+	proxyResourceList := &configv1.ProxyList{}
+	err := w.Client.List(context.TODO(), proxyResourceList)
```

Comment:
There's already some context available in

https://github.com/opendatahub-io/kubeflow/blob/7401baff76e07f87b75a9ae4bdbb69431144722f/components/odh-notebook-controller/controllers/notebook_webhook.go#L250-L251

why don't you pass it in as parameter and use that, instead of creating a TODO context?

Reply:
also very good point, I am still learning about all the ins and outs / details of odh notebook controller and the possibilities in golang. For here, I was merely adding back @VaishnaviHire code from way back then.